src/ui.js

```diff
@@ -40,200 +42,215 @@ the attractions of the terrain and the encounters they find there.
 <p>Code is available <a href="https://github.com/erik/derive">on GitHub</a>.</p>
 `,
 
-    export: `
+    exportImage: `
 <h3>Export Image</h3>
 
 <form id="settings">
-  <div class="form-row">
-    <label>Format:</label>
-    <select name="format">
-      <option selected value="png">PNG</option>
-      <option value="svg">SVG (no background map)</option>
-    </select>
-  </div>
-
-  <div class="form-row">
-    <label></label>
-    <input id="render-export" type="button" value="Render">
-  </div>
+    <div class="form-row">
+        <label>Format:</label>
+        <select name="format">
+            <option selected value="png">PNG</option>
+            <option value="svg">SVG (no background map)</option>
+        </select>
+    </div>
+
+    <div class="form-row">
+        <label></label>
+        <input id="render-export" type="button" value="Render">
+    </div>
 </form>
 
 <div id="export-container">
-  <ul id="export-list"></ul>
+    <ul id="export-list"></ul>
 </div>
 `
 }
 
 
 // Adapted from: http://www.html5rocks.com/en/tutorials/file/dndfiles/
 function handleFileSelect(map, evt) {
-    evt.stopPropagation()
-    evt.preventDefault()
-
-    let tracks = []
-    let files = evt.dataTransfer.files
-    let modal = buildUploadModal(files.length)
-
-    modal.show()
-
-    let fileIndex = 0
-    let parseFailures = []
-
-    function loadNextFile() {
-        if (fileIndex >= files.length) {
-            tracks.forEach(t => {
-                console.log('Adding track:', t.name)
-                map.addTrack(t.points)
-            })
-
-            if (parseFailures.length > 0) {
-                console.error('Failed files:', parseFailures)
-                window.alert(`Finished loading with ${parseFailures.length} failure(s)\n
-View console for info.`)
-            }
-
-            return modal.destroy()
+    evt.stopPropagation();
+    evt.preventDefault();
+
+    let tracks = [];
+    let files = evt.dataTransfer.files;
+    let modal = buildUploadModal(files.length);
+
+    modal.show();
+
+    let i = 0;
+    let parseFailures = [];
+
+    Promise.all(Array.prototype.map.call(files,file => new Promise(resolve => {
```

Comment:
Out of curiosity, why not `files.map(file => { ....})`?

Reply:
Because files is not an array, it's a [FileList](https://developer.mozilla.org/en-US/docs/Web/API/FileList), so it doesn't have the `map` method.

Reply:
Ahh. 👍